apps/publish/content/common.py

```diff
@@ -895,15 +895,40 @@ def _publish_associated_items(self, original, updates=None):
                         sync_associated_item_changes(associated_item, associated_item_updates)
                         continue
 
-                    if association_updates.get("state") not in PUBLISH_STATES:
-                        # There's an update to the published associated item
+                    orig_associated_item = archive_service.find_one(req=None, _id=associated_item[config.ID_FIELD])
+                    if self.is_changed(orig_associated_item, associated_item):
                         remove_unwanted(association_updates)
                         publish_service.patch(id=associated_item[config.ID_FIELD], updates=association_updates)
 
             # When there is an associated item which is published, Inserts the latest version of that associated item into archive_versions.
             insert_into_versions(doc=associated_item)
         self._refresh_associated_items(original)
 
+    def is_changed(self, old: dict, new: dict) -> bool:
+        """
+        Compare the associated items using SCHEMA fields, or fallback to SIGNIFICANT_FIELDS.
+        Return True if any significant field has changed.
+        """
+        SIGNIFICANT_FIELDS = {"headline", "slugline", "byline", "description_text"}
+
+        if len(old) != len(new):
+            return True
+
+        item_type = old.get("type") or new.get("type")
+        schema_config = app.config.get("SCHEMA", {})
+        schema_fields = set(schema_config.get(item_type, {}).keys())
```

Comment:
we are not really setting this anymore, it's part of respective content profile. but I would just check everything besides fields starting with `_`